src/components/Debugger/Timeline.tsx

```diff
@@ -264,6 +264,18 @@ const getPageUrl = (entries: Entry[]) => {
   return page ? new URL(page.value) : null;
 };
 
+const extractNetworkUserId = (cookies: Cookie[]): Cookie | undefined => {
+  // consider only cookies with the UUID format
+  const uuidRegexp =
+    /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/gi;
+  const uuidCookies = cookies.filter((x) => uuidRegexp.test(x.value));
+  // prefer a cookie with the name `sp` or take the first one
+  return (
+    uuidCookies.find((x) => x.name === "sp") ??
+    (uuidCookies.length > 0 ? cookies[0] : undefined)
```

Comment:
Should this be `uuidCookies[0]`? They'll basically be equivalent since we don't really expect more than the expected cookie to be sent but I guess it's theoretically possible with GTMSS or something to set more than the NWUID cookie in the response; we probably want the validated one to be certain, right?

We can probably then simplify to just `return uuidCookies.find((x) => x.name === "sp") ?? uuidCookies.shift();` which should already return `Cookie | undefined`.

Reply:
That was a typo, good catch!